tests/helper/skyMapTestCase.py

```diff
@@ -334,6 +334,33 @@ def getIndices(numItems):
                     self.assertBBoxPolygonOk(polygon=patchInfo.getOuterSkyPolygon(tractWcs=wcs),
                                              bbox=patchInfo.getOuterBBox(), wcs=wcs)
 
+    def testDm14809a(self):
+        """Generic version of test that DM-14809 has been fixed (first part)
+
+        The observed behaviour was:
+
+            skyMap.findTract(skyMap[9712].getCtrCoord()).getId() != 9712
+
+        We will generalise this over the entire skymap.
+        """
+        skymap = self.getSkyMap()
+        for tract in skymap:
+            self.assertEqual(skymap.findTract(tract.getCtrCoord()).getId(), tract.getId())
+
+    def testDm14809b(self):
+        """Generic version of test that DM-14809 has been fixed (second part)
+
+        The observed behaviour was:
+
+            skyMap[1].getCtrCoord() == skyMap[11].getCtrCoord()
+
+        We will generalise this over the entire skymap.
+        """
+        skymap = self.getSkyMap()
+        centers = set([(coord.getRa().asRadians(), coord.getDec().asRadians()) for
+                       coord in (tract.getCtrCoord() for tract in skymap)])
+        self.assertEqual(len(centers), len(skymap))
```

Comment:
Thank you very much for the new tests. However I worry that this test is not reliable in that the tiniest roundoff error in computing tract centers would allow the original problem to be undetected.

One possible solution is to round the recorded RA and Dec to some reasonable amount. A value based on neighborAngularSeparation would be ideal, but a fixed value such as 1e-4 radians would probably suffice.

Reply:
This isn't testing the centers, so I don't understand your concern. So long as the centers are contained exclusively within the tract (which I think is a strongly desirable property of all skymaps), this test works for what it's intended to prevent.